### PR TITLE
Move TraceSourceLoggerProvider to Microsoft.Framework.Logging.TraceSource

### LINKED ISSUE #192: `TraceSource` types in the wrong namespace
The `TraceSource` logger types are in the root `Microsoft.Framework.Logging` namespace.

These types should be in `Microsoft.Framework.Logging.TraceSource` to be consistent with the other loggers:

 - The `Console` types are in the `Microsoft.Framework.Logging.Console` namespace.
 - The `Debug` types are in the `Microsoft.Framework.Logging.Debug` namespace.
 - The `NLog` types are in the `Microsoft.Framework.Logging.NLog` namespace.

### PR DESCRIPTION
**Pull request by justinvp**
For consistency with the other loggers' namespaces.

Fixes #192

### CODE REVIEW

File: src/Microsoft.Framework.Logging.TraceSource/TraceSourceLogger.cs [hunk 1]
```diff
@@ -3,15 +3,17 @@
 
 using System;
 using System.Diagnostics;
-using Microsoft.Framework.Logging.Internal;
+using Microsoft.Framework.Logging.TraceSource.Internal;
 
-namespace Microsoft.Framework.Logging
+namespace Microsoft.Framework.Logging.TraceSource
 {
+    using DiagnosticsTraceSource = System.Diagnostics.TraceSource;
```
Comment (davidfowl) [inline src/Microsoft.Framework.Logging.TraceSource/TraceSourceLogger.cs]: This kinda sucks.

Comment (justinvp) [inline src/Microsoft.Framework.Logging.TraceSource/TraceSourceLogger.cs]: I know. There's no way around it because of `TraceSource` in the namespace.

Comment (Eilon) [inline src/Microsoft.Framework.Logging.TraceSource/TraceSourceLogger.cs]: Can this at least go next to the other `using`s?

Comment (justinvp) [inline src/Microsoft.Framework.Logging.TraceSource/TraceSourceLogger.cs]: Sure. I was just using the style I saw in some of the other repos.

**Comment by justinvp** [comment]
@Eilon, I updated the commit, moving the `using DiagnosticsTraceSource = ...` next to the other `using`s.

Not a big deal, but do you want to address other places where this style is used? I can submit bugs and/or PRs. A few examples:
 - https://github.com/aspnet/HttpAbstractions/blob/8703e2d7f21d09b50d20cc764e11d6bb0268aad2/src/Microsoft.AspNet.Http.Abstractions/Extensions/MapWhenExtensions.cs#L12
 - https://github.com/aspnet/HttpAbstractions/blob/eb0fe6a92ada31440e6db44d23b64f4d3d7240c0/src/Microsoft.AspNet.Owin/WebSockets/WebSocketAcceptAdapter.cs#L13-L29
 - https://github.com/aspnet/HttpAbstractions/blob/5fe8037281bb826e0708abdcdafbc76571dc21f5/src/Microsoft.AspNet.Owin/OwinEnvironment.cs#L24-L30

**Comment by Eilon** [comment]
@justinvp I think those are slightly different cases because those are inventing types out of the ether, as opposed to just working around a naming conflict. But either way, not super worth it :smile:

**Pull request merged**